src/lib/buffer.jl

```diff
@@ -45,8 +45,17 @@ _pullback(cx::AContext, ::typeof(Broadcast.materialize!), b::Buffer, x::Abstract
   _pullback(cx, copyto!, b, x)
 
 @adjoint function copy(b::Buffer)
-  copy(b), function (b̄)
-    grad_mut(__context__, b)[:] = b̄
+  res = copy(b)
+
+  function copy_sensitivity(b̄)
+    grad_mut(__context__, b)[:] .= vec(b̄)
     return
   end
+
+  function copy_sensitivity(b̄::Union{Tuple,Vector{T}}) where {T<:AbstractArray}
```

Comment:
```suggestion
  function copy_sensitivity(b̄::Union{Tuple,AbstractVector{T}}) where {T<:AbstractArray}
```

Fwiw,  `Buffer` mimics `similar` more than it does a literal array; and `similar(::Tuple)` would be an error.